flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -375,7 +374,7 @@ public JobSubmissionResult submitJob(JobGraph jobGraph, ClassLoader classLoader)
 	public void cancel(JobID jobID) throws Exception {
 		JobCancellationMessageParameters params = new JobCancellationMessageParameters();
 		params.jobPathParameter.resolve(jobID);
-		params.terminationModeQueryParameter.resolve(Collections.singletonList(TerminationModeQueryParameter.TerminationMode.CANCEL));
+		params.terminationModeQueryParameter.resolve(Collections.singletonList("CANCEL"));
```

Comment:
I liked using the enum `TerminationModeQueryParameter.TerminationMode` better. The deprecation of `TerminationModeQueryParameter` and `TerminationMode` was wrong and should be reverted. What should be deprecated is `TerminationMode.STOP` since `TerminationMode.CANCEL` is still valid for non-checkpointed and batch jobs.